pkg/controller/datavolume-controller.go

```diff
@@ -340,6 +340,26 @@ func addDatavolumeControllerWatches(mgr manager.Manager, datavolumeController co
 		}
 	}
 
+	// Watch for SC updates and reconcile the DVs waiting for default SC
```

Comment:
Can we use an indexer for this? The idea is very similar to this unit test: https://github.com/kubernetes-sigs/controller-runtime/blob/7e7bf8c1e1df991b957dfaa8b36d975740f309ab/pkg/cache/cache_test.go#L1182-L1210
We get the indexer for the client from mgr, define the indexing function, and then use the index when listing. Should look something like this:
```go
indexer := mgr.GetFieldIndexer()
indexFunc := func(obj client.Object) []string {
  return []string{string(obj.(*cdiv1.DataVolume).Status.Phase)}
}
indexer.IndexField(context.TODO(), &cdiv1.DataVolume{}, "status.phase", indexFunc)

if err := datavolumeController.Watch(&source.Kind{Type: &storagev1.StorageClass{}}, handler.EnqueueRequestsFromMapFunc(
  func(obj client.Object) (reqs []reconcile.Request) {
    dvList := &cdiv1.DataVolumeList{}
    if err := mgr.GetCache().List(context.TODO(), dvList, client.MatchingFields{"status.phase": ""}); err != nil {
      return
    }
    for _, dv := range dvList.Items {
      reqs = append(reqs, reconcile.Request{NamespacedName: types.NamespacedName{Name: dv.Name, Namespace: dv.Namespace}})
    }
    return
  },
),
); err != nil {
  return err
}
```
Might have to configure the indexer before you start the mgr not 100% sure.